exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -287,8 +284,10 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 		if err != util.ErrUnstructuredFieldNotFound {
 			return ctrl.Result{}, errors.Wrapf(err, "failed to retrieve replicas from infrastructure provider for MachinePool %q in namespace %q", mp.Name, mp.Namespace)
 		}
-	} else if mp.Status.Replicas == 0 {
-		log.Info("Retrieved unset Status.Replicas from infrastructure provider")
+	}
+
+	if len(providerIDList) == 0 && mp.Status.Replicas != 0 {
+		log.Info("Retrieved empty Spec.ProviderIDList from infrastructure provider but Status.Replicas is not zero.", "replicas", mp.Status.Replicas)
```

Comment:
I'm not sure we should log this, given that I expect this being a normal condition while machine are bootstrapping

Reply:
I'm not sure about this, the provider implementations ([AWS](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/controllers/awsmachinepool_controller.go#L273), [Azure](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d0b9621c43faca25266182fcfdcf585079688f06/azure/scope/machinepool.go#L222)) usually update `status.replicas` and `providerIDList` at the same time. 

Looking at the Azure implementation specificly: during instance boot, `len(providerIDList)` would be greater zero but `Status.Replicas == 0` and therefore wouldn't enter the condition.